src/account.ts

```diff
@@ -0,0 +1,843 @@
+            // Respond to the request normally, not with an error:
+            // Peer may have deposited to the channel and think it's confirmed, but we don't see it yet, causing them to send
+            // a claim value greater than the amount that's in the channel
+            return []
+          } else if (claimIncrement.lt(0)) {
+            throw new Error(`claim value of ${format(claim.value, Unit.Wei)} is less than previous claim for ${format(oldClaimValue, Unit.Wei)}`)
+          }
+
+          // If no channel is linked to this account, perform additional validation
+          if (!oldClaim) {
+            // Each channel key is a mapping of channelId -> accountName to ensure no channel can be linked to multiple accounts
+            // No race condition: a new linked channel will be cached at the end of this closure
+            const channelKey = `${claim.channelId}:incoming-channel`
+            await this.master._store.load(channelKey)
+            const linkedAccount = this.master._store.get(channelKey)
+            const canLinkChannel = !linkedAccount || linkedAccount === this.account.accountName
+            if (!canLinkChannel) {
+              throw new Error(`channel ${claim.channelId} is already linked to a different account`)
+            }
+
+            // Check the channel is to the server's address
+            // (only check for new channels, not per claim, in case the server restarts and changes config)
+            const amReceiver = channel.receiver.toLowerCase() === this.master._ethereumAddress.toLowerCase()
+            if (!amReceiver) {
+              throw new Error(`the recipient for incoming channel ${claim.channelId} is not ${this.master._ethereumAddress}`)
+            }
+
+            // Confirm the settling period for the channel is above the minimum
+            const isAboveMinSettlementPeriod = channel.settlingPeriod.gte(this.master._minIncomingSettlementPeriod)
+            if (!isAboveMinSettlementPeriod) {
+              throw new Error(`channel ${channel.channelId} has settling period of ${channel.settlingPeriod} blocks,`
+                + `below floor of ${this.master._minIncomingSettlementPeriod} blocks`)
+            }
+
+            // Deduct the initiation fee (e.g. this could be used to cover the cost of the claim tx to close the channel)
+            claimIncrement = claimIncrement.minus(this.master._incomingChannelFee)
+            if (claimIncrement.lt(0)) {
+              throw new Error(`claim value of ${format(newClaimValue, Unit.Wei)} is insufficient to cover the initiation fee of ` +
+                `${format(this.master._incomingChannelFee, Unit.Wei)}`)
+            }
+
+            this.master._store.set(channelKey, this.account.accountName)
+            this.master._log.trace(`Incoming channel ${claim.channelId} is now linked to account ${this.account.accountName}`)
+
+            // TODO web3.js beta 36 event bugs: https://github.com/ethereum/web3.js/issues/1916
```

Comment:
This seems kind of important, no?

Reply:
The channel watcher polls at regular intervals, so it still works fine. The event subscriptions are just nice because it'd trigger it to claim the channel instantly (if it was profitable). I'll add it back in after the Web3 bug is fixed.